src/Cardano/Wallet/Kernel/Wallet.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.Wallet.Kernel.Wallet where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Cardano.Wallet.Primitive
+    ( Coin )
+import Crypto.Encoding.BIP39
+    ( ValidMnemonicSentence )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import Fmt
+    ( Buildable (..), fmt )
+
+-- | Errors
+data CreateWalletError = CreateWalletError
+
+instance Buildable CreateWalletError where
+    build CreateWalletError =
+        mempty
+        <> "CreateWalletError"
+
+instance Show CreateWalletError where
+    show = fmt . build
+
+data GetWalletError = GetWalletError
+
+instance Buildable GetWalletError where
+    build GetWalletError =
+        mempty
+        <> "GetWalletError"
+
+instance Show GetWalletError where
+    show = fmt . build
+
+
+-- | Types
+newtype CreateWallet = CreateWallet NewWallet
```

Comment:
What's the rational for this newtype :thinking:  ?

Reply:
I started with **data** but then got :

Reply:
now added : 
```
data CreateWallet = CreateWallet NewWallet | ImportWallet
```

Reply:
Haha, got it. I was intrigued by the need for an extra type though (regardless of the syntax used) ? What's wrong with using `NewWallet` directly? I believe it echoes a previous remark I made about getting a `WalletLayer` type, and I suspect that this was the premise of it :sweat_smile:  ?